src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -146,6 +152,30 @@ private static int TryGetSelectedIndexFromDeclaration(int position, SyntaxNode m
             return parameters != null ? GetParameterIndex(parameters.Parameters, position) : 0;
         }
 
+        // Find the position to insert the new parameter.
+        // We will insert a new comma and a parameter.
+        protected override int? TryGetInsertPositionFromDeclaration(SyntaxNode matchingNode)
+        {
+            var parameters = matchingNode.ChildNodes().OfType<BaseParameterListSyntax>().SingleOrDefault();
+
+            if (parameters == null)
+            {
+                return null;
+            }
+
+            switch (parameters)
+            {
+                case ParameterListSyntax parameterListSyntax:
+                    return parameterListSyntax.CloseParenToken.SpanStart;
```

Comment:
What happens if these tokens are missing?

Reply:
Tried it in C#, and it miraculously works still. The token will be missing but still have an accurateish SpanStart. Completion works as expected in the Add Parameter dialog. Maybe we should fill in these things to make sure it's okay before continuing, or outright reject signatures of this form ahead of time.

Reply:
Filed https://github.com/dotnet/roslyn/issues/41835